### ISSUE TITLE
Filtering elements where value is null in Transformer

### ISSUE THREAD

**Issue by kickthemooon** [thephpleague/fractal#458]
Example:

```
return [
  'name' => null
]
```

This element is not filtered out. Is there some way tell Fractal to filter out null elements?

**Comment by willishq** [comment]
It'll have to be done manually, this is not really in the scope of Fractal.

If a value is meant to be in the response but there is no value for it, the property should still be there but not have a value in my opinion, otherwise you're putting responsibility on your consumer to check if the keys are there in the response, which shouldn't happen.

Imagine every property on the object needing the API consumer to check if it exists?

**Comment by johannesschobel** [comment]
i totally agree with @willishq .. it is better to have a "stable interface / structure" for the return types of a transformer..

however, if you want to filter out some properties that are explicitly set to `null` you can use the `array_filter()` method.. please note that this function may also remove `falsy` values, like `false` or `''`..

I think, this issue can be closed @kickthemooon

**Comment by hubertnnn** [comment]
I think both ways are valid.
I prefer keeping the key, but I can see why some people might want to remove null values from response (eg. to reduce network usage or hide some fields).
But for that the best way is to create a custom serializer (or modify existing one).
The default should still be keeping the keys.

Also maybe having a way to say that something does not exist (as opposed to being empty) might be a solution.
So adding a new keyword like this:

```
protected $defaultIncludes = [ 'a',  'b',  'c'];

public function includeA($item)
{
        return $this->primitive(1);
}

public function includeB($item)
{
        return $this->null();
}

public function includeC($item)
{
        return $this->skip();
}
```

To return
```
{
  "a": 1,
  "b": null
}
```
